### PR TITLE
SONARJAVA-1228 Add new JavaVersionAwareVisitor into api

### PR DESCRIPTION
**Pull request by m-g-sonar**
Rules can now implements interface `org.sonar.plugins.java.api.JavaVersionAwareVisitor` in order to react to java version.

### CODE REVIEW

File: sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSquidSensor.java [hunk 1]
```diff
@@ -51,6 +53,8 @@
 @DependedUpon(value = JavaUtils.BARRIER_AFTER_SQUID)
 public class JavaSquidSensor implements Sensor {
 
+  private static final Logger LOG = LoggerFactory.getLogger(JavaSquidSensor.class);
```
Comment (SonarTech) [inline sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSquidSensor.java]: MAJOR Remove this unused "LOG" private field. [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1068)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by benzonico** [comment]
Before proceeding any further please keep in mind this rule : http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1190 which you won't be able to cover with your current approach.

**Comment by m-g-sonar** [comment]
Good point @benzonico. We will look back at it to be able to handle such cases correctly.

**Pull request merged**